FrEee/Game/Objects/LogMessages/LogMessage.cs

```diff
@@ -33,5 +35,24 @@ protected LogMessage(string text, int turn)
 		/// The turn number on which the log message was generated.
 		/// </summary>
 		public int TurnNumber { get; set; }
+
+		/// <summary>
+		/// The type of this message. 
+		/// </summary>
+		public LogMessageType LogMessageType { get; set; }
+
+		
+	}
+	public enum LogMessageType

```

Comment:
So these log message types - I'm not too keen on having a hardcoded list of types; it would be nice if you could use an arbitrary string, or have each type be a subclass of LogMessage so you can add more in a modular fashion. This is for the AI, right? So they can understand log messages better? If so, it might be best to have subclasses so each type of message can store its own custom data. But if you think an enum would work better, that's OK.

Reply:
Actually, I was thinking you could use them for the client as well, as a checkbox list to hide and show on the Log View. So each turn you can easily find all new colonies, new construction, and the like. Or hide things like errors, warnings, and other things you don't want to deal with. 

It might work better as a string, but this is more for sorting than telling the AI what the log message is for.

Reply:
Oh, that would be cool, being able to filter the log. What do you mean by sorting, though? Why not create a separate class for each type of log message, or if that's too much, just use string constants?

Reply:
I meant filtering than sorting. Brain mixing up words. 

And do we need a seperate class for each log message type? Pictatorial is working for most things well enough. I suppose it might be advantageous to have seperate classes, but my aim was to get the AI some way to tell something had happened.

Reply:
I'm not sure, I'm just thinking different types of log messages might have different data associated with them which we could then access in a script via those classes, rather than having to do everything as a generic log message.

Reply:
I can think of a location log message format, that would have a sector associated with it (zoom when you click) or the research log message would open the research screen (if they don't already). But really, what data do we need that we don't have currently via the context?

Reply:
Hmm, you think all the data we might need is already present in the context? OK then, I guess we can go ahead and use the enum...